### ISSUE TITLE
Issues using "Right click > Rebase current branch onto branch" with branches created before a recent update?

### ISSUE THREAD

**Issue by aminomancer** [gitkraken/vscode-gitlens#3447]
### Description

1. Be on an old feature branch - this only seems to happen with branches I created a few weeks ago, presumably before a gitlens or vscode update? If I create a new branch many commits behind main and try to rebase that, it works fine. But if I try to use this context menu item to rebase branches that I created a few weeks ago (or older), the following steps play out.
2. Right click the main branch
3. Select "Rebase current branch onto branch..."
4. The dropdown menu that appears from the Command Center shows 2 options, Rebase and Interactive Rebase
5. Click Interactive Rebase to see what commit that is
6. It looks alright... it's the current, unpushed commit(s), above the head
7. Following through with the rebase will put all the commits from main ON TOP of your feature commit, without altering the diff(s) of the unpushed commits in the rebased branch

So the rebase context menu item is basically working backwards for these branches, for whatever reason.

I checked that this is not an issue with vscode or git. I tried rebasing using vscode's `git.rebase` command ("Git: Rebase branch..." in the command palette) and it works just fine. I also tried rebasing from the git CLI and it worked fine.

Unfortunately there's only so much testing I can do since I only have a finite number of old branches to play with, and the super mega old ones are just too tedious to keep checking out back and forth.

There weren't any warnings in the GitLens output, but I tried setting the log level to `debug`, and I have attached that debug output. It's very long. The GitLens (git) output looked normal and very short in comparison.

### GitLens Version

v2024.8.305 (pre-release)

### VS Code Version

Version: 1.92.0 (user setup)
Commit: b1c0a14de1414fcdaa400695b4db1c0799bc3124
Date: 2024-07-31T23:26:45.634Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.19045

### Git Version

git version 2.43.0.windows.1

### Logs, Screenshots, Screen Captures, etc

I'm attaching the logs as files since one of them is very large.

[GitLens output.txt](https://github.com/user-attachments/files/16503095/GitLens.output.txt)
[GitLens (git) output.txt](https://github.com/user-attachments/files/16503094/GitLens.git.output.txt)

**Comment by eamodio** [comment]
Unfortunately, I can't fathom what could be going on here and the attached logs don't show a rebase operation at all. Is there something "special" about those old branches in your `.git/config` file?

**Comment by aminomancer** [comment]
> Unfortunately, I can't fathom what could be going on here and the attached logs don't show a rebase operation at all.

Hmm... maybe I did something wrong generating the log. I'll try another one and get back to you.

> Is there something "special" about those old branches in your `.git/config` file?

I initially thought this was the case, because I did notice a difference between the 3 week old branches and the test branch I created for purposes of testing. This gets a little complicated, as I'm using [git-cinnabar](https://github.com/glandium/git-cinnabar) which sort of translates between mercurial and git. Here are some excerpts from my local repo config:

```
[remote "origin"]
  url = hg::https://hg.mozilla.org/mozilla-unified
  fetch = +refs/heads/*:refs/remotes/origin/*
[branch "central"]
  remote = origin
  merge = refs/heads/bookmarks/central
  vscode-merge-base = origin/branches/default/tip

[branch "a3WeekOldBranch"]
  vscode-merge-base = origin/bookmarks/central
[branch "rebaseTestBranch"]
  vscode-merge-base = origin/branches/default/tip
```

At first I thought the difference in `vscode-merge-base` values was the issue here, since it was the only notable difference between the branches, other than when and under what circumstances they were created. But after further testing, I think this connection is spurious.

I found that, when I create a branch using the + button next to the title of the Branches view (the most common way for me to create branches), it sets `vscode-merge-base` to `origin/bookmarks/central`. But when I create a branch by right-clicking a commit > "Create Branch...", it sets `vscode-merge-base` to `origin/branches/default/tip`. So it just happens that when creating this `rebaseTestBranch`, the simplest approach was to create a branch from an old commit, which for whatever reason gives a different `vscode-merge-base`. And I don't think it even makes any difference, I think either value basically means the same thing, where the bookmarks version is just the hg equivalent of branches/default/tip.

So, excluding that, the only difference is that the branches that have problems rebasing are at least a few weeks older, and I think there have been a couple vscode updates and several GitLens updates (I'm on the pre-release channel) since then. Since I created those branches with GitLens, maybe it created them differently in a way that can't be seen in the config file?
